model/common/src/icon4py/model/common/decomposition/definitions.py

```diff
@@ -365,4 +379,20 @@ def create_single_node_exchange(
     return SingleNodeExchange()
 
 
+@functools.singledispatch
+def create_global_reduction(props: ProcessProperties) -> Reductions:
+    """
+    Create an Exchange depending on the runtime size.
+
+    Depending on the number of processor a SingleNode version is returned or a GHEX context created and a Multinode returned.
+    """
+    raise NotImplementedError(f"Unknown ProcessorProperties type ({type(props)})")
+
+
+@create_global_reduction.register(SingleNodeProcessProperties)
+def create_global_reduction_exchange(props: SingleNodeProcessProperties) -> Reductions:
+    return SingleNodeReductions()
```

Comment:
you need to handle the `array_ns` here somehow.... or you remove it from the constructor and pass it to the `min` function to post pone the decision here, where we don not really know. Wherever we call the `min` there probably is a `backend` around to determine the device.